rfc/previews-using-fetaure-flags.md

```diff
@@ -0,0 +1,306 @@
+    if: github.event_name == 'synchronize'
+    steps:
+      - uses: wundergraph/feature-flags@main
+        with:
+          config: cosmo.yaml # Default to .github/cosmo.yaml
+          update: true # Only update the feature subgraphs from cosmo.yaml
+      - run: |
+          echo "Feature subgraph has been updated. You can now update the subgraph service."
+          echo "Published Feature Subgraphs: ${{ steps.ff.outputs.published_feature_subgraphs }}"
+
+  destroy:
+    if: github.event.pull_request.action == 'closed'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: wundergraph/feature-flags@main
+        id: ff
+        with:
+          config: cosmo.yaml # Default to .github/cosmo.yaml
+          destroy: true # Only destroy the feature flag + subgraphs from cosmo.yaml
+      - run: |
+          echo "Feature flag and the feature subgraphs have been destroyed. You can now unprovision the subgraph service."
+---
+# config
+# .github/cosmo.yaml
+namespace: 'staging'
+feature_flags:
+  - name: 'my-feature-flag-{PR_NUMBER}'
+    labels:
+      - 'my-feature-flag-label'
+
+subgraphs:
+  - name: 'subgraph1'
+    schema: 'my-project/my-subgraph/subgraph1.graphql'
+    routing_url: 'http://mycompany-subgraph1-{PR_NUMBER}.com/graphql' # Supports templating, only PR_NUMBER is supported
+```
+
+### Requirements
+
+- Make sure the labels of the feature flag match that of the federated graph for which the preview is being created.
+- Make sure that the subgraphs being changed are a part of the federated graph.
+- Make sure that the changed subgraphs are a part of the config, if not present the featur subgraph for that subgraph will not be created.
+- All the subgraphs in the config have to be deployed and present on Cosmo.
+- The `cosmo.yaml` file should be present in the `.github` directory on the first commit of the PR(if already not present).
+- The changes to the `cosmo.yaml` file should be in the first commit of the PR.
+- All the subgraphs changed in this PR should be deployed at the proposed routing_url in the CI for the preview to work.
+
+### Implementation
+
+#### Create Job
+
+- This job will create a new feature flag and create/publish the feature subgraphs for all the changed subgraphs.
+- This job will only run when the PR is opened or reopened, so it is important for the `cosmo.yaml` file to be present in the `.github` directory on the first commit of the PR.
+
+#### Update Job
+
+- This job will publish(create if not present) the feature subgraphs for all the changed subgraphs in the latest commit.
+- This job will only run on each new commit to the PR.
+
+#### Destroy Job
+
+- This job will destroy the feature flag and the feature subgraphs used for the preview of this PR.
+- This job will only run when the PR is closed/merged.
+
+
+### Limitations
+
+- The comso.yaml file should not be changed after the PR is opened, if changed for the preview to work as expected, the PR has to be closed and reopened.
```

Comment:
Is it possible to detect this and raise a CI error, telling the user to close and re-open the PR?
or, maybe even easier, if you detect this, delete the FF and recreate it?

Reply:
We should be able to provide that error.

Regarding deleting the FF, we will have to do a little more sometimes, like if they change the namespace then we will have to delete everything and then provision everything.

Reply:
This sounds dangerous.
I think it's better to just raise an error.